suite-native/device-manager/src/components/DeviceInfoButton.tsx

```diff
@@ -0,0 +1,72 @@
+import { useSelector } from 'react-redux';
+import { Pressable } from 'react-native';
+
+import { useNavigation } from '@react-navigation/native';
+
+import { analytics, EventType } from '@suite-native/analytics';
+import { HStack, Text } from '@suite-native/atoms';
+import { selectDevice, selectIsDeviceDiscoveryActive } from '@suite-common/wallet-core';
+import { Translation } from '@suite-native/intl';
+import {
+    RootStackParamList,
+    RootStackRoutes,
+    StackToStackCompositeNavigationProps,
+} from '@suite-native/navigation';
+import { Icon } from '@suite-common/icons';
+import { FeatureFlag, useFeatureFlag } from '@suite-native/feature-flags';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+
+import { useDeviceManager } from '../hooks/useDeviceManager';
+import { DeviceAction } from './DeviceAction';
+
+type NavigationProp = StackToStackCompositeNavigationProps<
+    RootStackParamList,
+    RootStackRoutes.AppTabs,
+    RootStackParamList
+>;
+
+const flexStyle = prepareNativeStyle<{ showWithFlex: boolean }>((_, { showWithFlex }) => ({
+    extend: {
+        condition: showWithFlex,
+        style: {
+            flex: 1,
+            justifyContent: 'center',
+        },
+    },
+}));
+
+export const DeviceInfoButton = () => {
+    const { applyStyle } = useNativeStyles();
+    const navigation = useNavigation<NavigationProp>();
+    const { setIsDeviceManagerVisible } = useDeviceManager();
+    const [isPassphraseFeatureEnabled] = useFeatureFlag(FeatureFlag.IsPassphraseEnabled);
+    const selectedDevice = useSelector(selectDevice);
+    const isDiscoveryActive = useSelector(selectIsDeviceDiscoveryActive);
+    const showWithFlex =
```

Comment:
The name is not indicating very well why these properties are dependant on the flex style props. Could you elaborate more so we can figure out a better / more descriptive name so it's more clear when reading this component?

Reply:
For some occasions we want this element be full width and for some no. I might redo this if I put pressable into the DeviceAction element

Reply:
I understand, but IMO it should be clearer why this should be affected. Is it because sometimes it's rendered side by side with something etc. It's a nit, but when I would come to refactor this, it would be a pain to find out what this means